archive_upload/handlers/dsmc_handlers.py

```diff
@@ -680,11 +683,17 @@ def post(self, runfolder):
             except (NameError):
                 raise ArchiveException(reason=missing_rm_msg, status_code=400)
 
+        if "required_dirs" in request_data:
```

Comment:
I like the pattern:

```
required_dirs = request_data.get("required_dirs", None)
if required_dirs: ...
```

Then you only have to use the "required_dirs" literal once.

Reply:
Fixed